packages/diagram_capture/test/diagram_capture_test.dart

```diff
@@ -158,22 +160,35 @@ void main() {
       final DiagramController controller = new DiagramController(
         builder: (BuildContext context) => new TestAnimatedDiagram(key: key, size: 1.0),
         outputDirectory: outputDir,
-        frameFilenameGenerator: filenameGenerator,
+        filenameGenerator: filenameGenerator,
         pixelRatio: 1.0,
         screenDimensions: const Size(100.0, 100.0),
       );
 
       controller.builder = (BuildContext context) => new TestAnimatedDiagram(key: key, size: 50.0);
-      final List<File> outputFiles = await controller.drawAnimatedDiagramToFiles(
+      final File outputFile = await controller.drawAnimatedDiagramToFiles(
         end: const Duration(milliseconds: 1200),
         frameDuration: const Duration(milliseconds: 200),
       );
-      expect(outputFiles.length, equals(7));
-      expect(outputFiles[0].path.endsWith('test_name_0_0.png'), isTrue);
-      expect(outputFiles[6].path.endsWith('test_name_1200_6.png'), isTrue);
-      final List<int> expectedSizes = <int>[1, 11, 21, 31, 41, 50, 50];
+      expect(outputFile.path.endsWith('test_name.json'), isTrue);
       int count = 0;
-      for (File file in outputFiles) {
+      expect(outputFile.existsSync(), isTrue);
+      expect(outputFile.lengthSync(), greaterThan(0));
+
+      Map<String, dynamic> _loadMetadata(File metadataFile) {
+        final Map<String, dynamic> metadata = json.decode(metadataFile.readAsStringSync());
+        final String baseDir = path.dirname(metadataFile.absolute.path);
+        final List<File> frameFiles = metadata['frame_files']
+            .map<File>((dynamic name) => new File(path.normalize(path.join(baseDir, name)))).toList();
+        metadata['frame_files'] = frameFiles;
+        return metadata;
```

Comment:
i guess this is fine since it's a hacky little script but fwiw best practice would be to convert out of this JSONesque map and into fully typed structured data as quick as possible.

Reply:
OK, I'll build a little wrapper class.